test/cri-containerd/policy_test.go

```diff
@@ -116,3 +127,127 @@ func Test_RunSimpleAlpineContainer_WithPolicy_Allowed(t *testing.T) {
 	startContainer(t, client, ctx, containerID)
 	stopContainer(t, client, ctx, containerID)
 }
+
+func Test_RunContainers_WithSyncHooks_Positive(t *testing.T) {
+	requireFeatures(t, featureLCOW, featureLCOWIntegrity)
+
+	type config struct {
+		name              string
+		waiterSideEffect  func(containerConfig *securitypolicy.ContainerConfig)
+		shouldError       bool
+		expectedErrString string
+	}
+
+	for _, testConfig := range []config{
+		{
+			name:             "ValidWaitPath",
+			waiterSideEffect: nil,
+			shouldError:      false,
+		},
+		{
+			// This is a long test that will wait for a timeout
+			name: "InvalidWaitPath",
+			waiterSideEffect: func(cfg *securitypolicy.ContainerConfig) {
+				cfg.ExpectedMounts = []string{"/mnt/shared/container-B/wrong-sync-file"}
+			},
+			shouldError:       true,
+			expectedErrString: "timeout while waiting for path",
```

Comment:
I feel like this test is made confusing by having these two scenarios since they're so different? That or I just don't understand what the test is doing 😅

Reply:
yeah, this might be confusing agree. let me see if I can split the test setup properly...